pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java

```diff
@@ -348,4 +349,25 @@ private static HttpRequesterIdentity makeHttpIdentity(org.glassfish.grizzly.http
 
     return identity;
   }
+
+  /**
+   * Generate Response object from the BrokerResponse object with 'X-Pinot-Error-Code' header value
+   * @param brokerResponse
+   * @return Response
+   * @throws Exception
+   */
+  private static Response getPinotQueryResponse(BrokerResponse brokerResponse)
+      throws Exception {
+    int headerValue = -1; // default value of the header.
+
+    if (brokerResponse.getExceptionsSize() != 0) {
+      // set the header value as first exception error code value.
+      headerValue = brokerResponse.getProcessingExceptions().get(0).getErrorCode();
+    }
+
+    // returning the Response with OK status and header value.
+    return Response.ok()
+        .header(PINOT_QUERY_ERROR_CODE_HEADER, headerValue)
```

Comment:
it is a bit weird to response "OK" then with an "ERROR code" IMO

CC @ankitsultana to take a look

Reply:
Yeah the issue is that the broker API returns exceptions in the JSON in most scenarios, with the HTTP Status Code set to 200. We can't change that behavior without making it backwards incompatible, hence the proposal was to add an additional response header.

Reply:
got it. will wait for a bit of feedback. was wondering if we should take similar naming with QueryException (which already have a number-based error code)

Reply:
@walterddr can you please explain more about similar naming with QueryException? Thanks

Reply:
https://github.com/apache/pinot/blob/9314c116adf0330a3c4b38d722f53a857a618add/pinot-common/src/main/java/org/apache/pinot/common/exception/QueryException.java#L49

Reply:
> or do we want to use a different header

In the long term if we also want to report whether an error is retryable, or a client error, then we would have to add a new header.

Ideally the error-codes in QueryException should have followed some convention (like all 4xx are user errors, and 5xx are internal errors, etc.).

But given the business logic is coded into Pinot itself, I don't think we have any choice other than converting the error-code into the appropriate flag in Pinot and returning it in a dedicated header.

e.g. for marking an error as a client-error, we could potentially add another header like `X-Pinot-Client-Error: true`.

Reply:
I agree. In the future, we can extend this header approach based on the use-case requirements. 
Also, as @ankitsultana mentioned, we should have a convention around error codes, but we need to figure out how end users use it and how we can introduce better error codes with backward compatibility.